### PR TITLE
Add pylint checker that disallows __del__ methods

### PR DESCRIPTION
**Pull request by ulope**
(Based on #5564, merge after)

## Description

`__del__` methods are notoriously unreliable and have caused multiple issues in the past (#5561, #4346, #3559).
We should avoid using them in the code base in general.

This adds a pylint plugin that adds two new checks:
- C6493: A `C`onvention level check that recommends against using `__del__` methods in general
- E6494 An `E`rror level check that warns against raising exceptions in `__del__` methods

### CODE REVIEW

File: tools/pylint/del_method_checker.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+from astroid import FunctionDef
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+from . import find_parent, ignore_tests
+
+ID_DEL_METHOD = "del-method-used"
+ID_EXCEPTION_IN_DEL = "exception-in-del"
```
Comment (karlb) [inline tools/pylint/del_method_checker.py]: Do we have to enable these in the pylintrc? We have a `disable=all` in there.

**Comment by hackaugusto** [comment]
btw, the use of `__del__` for the "linear type" was removed with https://github.com/raiden-network/raiden/pull/5977 , that PR made the usage of the `nonce` more strict (now transactions are always sent, so there is not need for the linear type)

**Comment by palango** [comment]
Rebased that one, @karlb still has a point: https://github.com/raiden-network/raiden/pull/5568#discussion_r378746221

File: tools/pylint/del_method_checker.py (hunk 1 above)
Comment (palango) [reply tools/pylint/del_method_checker.py]: Added a commit to do this.

File: pylintrc [hunk 2]
```diff
@@ -47,6 +47,8 @@ enable=
     unused-import,
     unused-variable,
     useless-object-inheritance,
+    del-method-used,
+    exception-in-del,
```
Comment (ulope) [inline pylintrc]: Please keep these sorted alphabetically. Not having them sorted makes it much too easy to overlook something. (e.g. how we had the assert checker accidentally disabled for nearly a year)

Comment (palango) [reply pylintrc]: Done

**Pull request merged**